resources/views/standings/upload.blade.php

```diff
@@ -1,241 +1,3533 @@
+                    constructorSelect += `<option value=${supportingVariables.availableConstructors[y].id}>${constructorDispVal}</option>`;
+                }
+                else {
+                    constructorSelect += `<option selected value=${supportingVariables.availableConstructors[y].id}>${constructorDispVal}</option>`;
+                }
+                constructorCol.innerHTML = constructorSelect;
+            }
+            constructorCol.dispatchEvent(e);
+        }
+    }
+
+    function checkAllTableValuesForErrors(json, regexValidationStrings, jsonResultsDetailsStore, supportingVariables) {
+        let postStatus = {
+            track: 1,
+            results: 1
+        }
+
+        checkEachCellValueForError(json, postStatus);
+
+        if(
+            isValidTimeFormat(json, regexValidationStrings, jsonResultsDetailsStore) && 
+            (postStatus.track === 1) && 
+            (postStatus.results === 1)
+        ) {
+            $('#errorSubmitJSONAlert').slideUp(500);
+            $('#submitJSON').removeClass('disableActionBtns');
+            $('#reviewJSON').removeClass('disableActionBtns');
+        }
+        else {
+            $('#submitJSON').addClass('disableActionBtns');
+            $('#reviewJSON').addClass('disableActionBtns');
+            $('#errorSubmitJSONAlert').html('<p>Please clear all the <strong>ERRORS</strong></p>');
+            $('#errorSubmitJSONAlert').slideDown(500);
+        }
+    }
+
+    function checkEachCellValueForError(json, postStatus) {
+        let trackOuterDivs = ['#trackBodySeason', '#trackBodyRound', '#trackBodyCircuit', '#trackBodyPoints'];
+
+        for(let i = 0; i < trackOuterDivs.length; i++) {
+            postStatus.track -= checkForError(trackOuterDivs[i]);
+        }
+
+        for(let i = 0; i < json.results.length; i++){
+            let resultsOuterDivs = [`#resultsBodyPos${i}`, `#resultsBodyDriver${i}`, `#resultsBodyConstructor${i}`, `#resultsBodyGrid${i}`, `#resultsBodyStops${i}`, `#resultsBodyFl${i}`, `#resultsBodyTime${i}`, `#resultsBodyStatus${i}`];
+            
+            for(let j = 0; j < resultsOuterDivs.length; j++) {
+                postStatus.results -= checkForError(resultsOuterDivs[j]);
+            }
+        }
+    }
+
+    function checkForError(value) {
+        let flag = 0;
+        
+        if($(value).hasClass('bg-red-600')) {
+            flag = 1;
+        }
+
+        return flag;
+    }
+
+    function isValidTimeFormat(json, regexValidationStrings, jsonResultsDetailsStore) {
+        let timeCheckFastestLap = new RegExp(regexValidationStrings.regexFastestLap);
+        let timeCheckAbsolute = new RegExp(regexValidationStrings.regexTimeIsAbsolute);
+        let timeCheckInterval = new RegExp(regexValidationStrings.regexTimeIsInterval);
+        let postStatus = 1;
+        
+        for(let i = 0; i < json.results.length; i++) {
```

Comment:
Bruh what is this monstoriety ???

Reply:
Noted. Will be simplified into a more readable code.